src/main/resources/js/colorPickMacro.js

```diff
@@ -0,0 +1,90 @@
+const baseUrl = AJS.params.baseUrl;
+const pageID = AJS.params.pageId;
+
+function updateImage(file, uniqueFilename) {
+    console.log(file);
+    var actionData = new FormData();
+    actionData.append('file', file, uniqueFilename);
+    actionData.append('comment', "foobar");
+    actionData.append('minorEdit', "true");
+    AJS.$.ajax({
+        url: baseUrl + '/rest/api/content/' + pageID + '/child/attachment',
+        type: "POST",
+        data: actionData,
+        dataType: "json",
+        processData: false,
+        headers: {
+            "X-Atlassian-Token": "nocheck"
+        },
+        contentType: false,
+        cache: false,
+        success: function (content) {
+            var alldata = content.results;
+            var lastImage = alldata[alldata.length - 1];
+            var link = lastImage._links.download;
+            var fullpath = baseUrl + link;
+            uploadDB(fullpath)
+        },
+        error: function () {
+            console.log("error while creating attachment")
+        }
+    })
+}
+
+function uploadDB(fullpath) {
+    AJS.$.ajax({
+        url: baseUrl + '/rest/restresource/1.0/attach/' + pageID,
+        type: "POST",
+        headers: {
+            "path": fullpath,
+            "X-Atlassian-Token": "nocheck"
+        },
+        success: function () {
+            setBackgroundImage(fullpath);
+        },
+        error: function () {
+            console.log("error while upload information in database")
```

Comment:
what if uploading of the information to the database will fail? maybe it will be good to delete the attached image to the confluence page?